chore: remove duplicate typescript example (#34449)
vercel/next.js: examples/with-typescript/README.md

# TypeScript Next.js example

> This examples is part of the `create-next-app` CLI as an official TypeScript template and can be used with the `--ts` or `--typescript` flag.

Reviewer: ```suggestion
This example is part of the `create-next-app` CLI as an official TypeScript template and can be used with the `--ts` or `--typescript` flag.
```

> You can find its source code here: https://github.com/vercel/next.js/tree/canary/packages/create-next-app/templates/typescript

Reviewer: ```suggestion
You can find the source code here: https://github.com/vercel/next.js/tree/canary/packages/create-next-app/templates/typescript
```

Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:

```sh
npx create-next-app --typescript with-typescript-app
# or
yarn create next-app --typescript with-typescript-app
```

Deploy it to the cloud with [Vercel](https://vercel.com/new?utm_source=github&utm_medium=readme&utm_campaign=next-example) ([Documentation](https://nextjs.org/docs/deployment)).

## Notes

> This example shows how to integrate the TypeScript type system into Next.js. Since TypeScript is supported out of the box with Next.js, all we have to do is to install TypeScript. (Installing this template will do this for you)

Reviewer: ```suggestion
This example shows how to integrate the TypeScript type system into Next.js. Since TypeScript is supported out of the box with Next.js, all you have to do is to get started is to use this example. Alternatively, you can [install TypeScript manually](https://nextjs.org/docs/basic-features/typescript#existing-projects) to an existing project. 
```

A suggestion, what we're trying to say is, you can get started with TypeScript by using this example or the CLI but you can also add TypeScript to your existing project - link to docs.

```sh
npm install --save-dev typescript
# or
yarn add --dev typescript
```

To enable TypeScript's features, we install the type declarations for React and Node.

```sh
npm install --save-dev @types/react @types/react-dom @types/node
# or
yarn add --dev @types/react @types/react-dom @types/node
```

> When we run `next dev` the next time, Next.js will start looking for any `.ts` or `.tsx` files in our project and builds it. It even automatically creates/updates a `tsconfig.json` file for our project with the recommended settings.

Reviewer: ```suggestion
When we run `next dev` the next time, Next.js will start looking for any `.ts` or `.tsx` files in your project and builds it. It also automatically creates and updates a `tsconfig.json` file for your project with the recommended settings.
```
I'm a little confused with the `builds it` bit, is it referring to the project or the files?

Author: Yeah, this one does sound a bit confusing. I think it refers to the `next dev` that will build (compile?) your TS files, but that's not necessary to mention here as it's an opaque step (you won't see/use the TS->JS output).

Author: > for any `.ts` or `.tsx`

Is also inaccurate, as it also looks for `.js` and `.jsx` files.

Author: I propose

> When we run `next dev` the next time, Next.js will also start looking for any `.ts` or `.tsx` files. It even creates or updates the `tsconfig.json` file for your project with the recommended settings.

Reviewer: Yes, I like that. It's clearer 👍🏼  Just realized the tense of the original paragraph is slightly off. Sorry, just a grammar nitpick: 

> When you run `next dev` the next time, Next.js will look for any `.ts` or `.tsx `files. It'll also create or update the `tsconfig.json` file for your project with the recommended settings.
